### DISCUSSION TITLE
Find out used transport

### DISCUSSION THREAD

**Discussion by RonnieRocket147** [socketio/socket.io#3839] [Q&A] [answered]
Is there a way to find out the active transport for a connected client socket?

Thank you!

**Comment by darrachequesne** [comment, marked as answer]
```js
const currentTransport: socket.connected ? socket.io.engine.transport.name : "-";
```

**Reply by RonnieRocket147** [reply]
Thank you!

**Reply by RonnieRocket147** [reply]
This works on the client side. Is it also possible to detect to transport of a socket on the server side?

**Reply by darrachequesne** [reply]
On the server-side:

```js
io.on("connection", socket => {
  const currentTransport = socket.conn.transport.name;

});
```

I don't think this is currently covered in the documentation, I will add it.
